From: Pedro Perez de Heredia <pedro.perez@digi.com>
Date: Tue, 7 Nov 2017 13:17:48 +0100
Subject: [PATCH] ARM: dts: ccimx6ulstarter: add support for Sense HAT

Device tree configuration for the Sense HAT board, on the ConnectCore 6UL
SBC Express.

https://www.raspberrypi.org/products/sense-hat/

Signed-off-by: Pedro Perez de Heredia <pedro.perez@digi.com>
---
 arch/arm/boot/dts/imx6ul-ccimx6ulstarter-id129.dts | 8 ++++++++
 arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts    | 8 ++++++++
 arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dts       | 8 ++++++++
 3 files changed, 24 insertions(+)

diff --git a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-id129.dts b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-id129.dts
index aad2fd7a..4784ed6 100644
--- a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-id129.dts
+++ b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-id129.dts
@@ -84,6 +84,14 @@
 /* I2C2 (Raspberry PI Expansion header and Grove connector) */
 &i2c2 {
 	status = "okay";
+	rpi-sense@46 {
+		reg = <0x46>;
+		compatible = "rpi,rpi-sense";
+		keys-int-gpios = <&gpio3 12 1>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <12 IRQ_TYPE_EDGE_RISING>;
+		status = "okay";
+	};
 };
 
 /* MCA IO0 connected to Grove ADC connector */
diff --git a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts
index aad2fd7a..4784ed6 100644
--- a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts
+++ b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter-wb.dts
@@ -84,6 +84,14 @@
 /* I2C2 (Raspberry PI Expansion header and Grove connector) */
 &i2c2 {
 	status = "okay";
+	rpi-sense@46 {
+		reg = <0x46>;
+		compatible = "rpi,rpi-sense";
+		keys-int-gpios = <&gpio3 12 1>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <12 IRQ_TYPE_EDGE_RISING>;
+		status = "okay";
+	};
 };
 
 /* MCA IO0 connected to Grove ADC connector */
diff --git a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dts b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dts
index 0a040a3..ce61ac0 100644
--- a/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dts
+++ b/arch/arm/boot/dts/imx6ul-ccimx6ulstarter.dts
@@ -84,6 +84,14 @@
 /* I2C2 (Raspberry PI Expansion header and Grove connector) */
 &i2c2 {
 	status = "okay";
+	rpi-sense@46 {
+		reg = <0x46>;
+		compatible = "rpi,rpi-sense";
+		keys-int-gpios = <&gpio3 12 1>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <12 IRQ_TYPE_EDGE_RISING>;
+		status = "okay";
+	};
 };
 
 /* MCA IO0 connected to Grove ADC connector */
